src/python/pants/backend/jvm/subsystems/jvm_platform.py

```diff
@@ -106,7 +136,7 @@ def _fallback_platform(self):
     source_level = JvmPlatform.parse_java_version(DistributionLocator.cached().version)
     target_level = source_level
     platform_name = f'(DistributionLocator.cached().version {source_level})'
-    return JvmPlatformSettings(source_level, target_level, [], name=platform_name)
+    return JvmPlatformSettings(source_level, target_level, [], [], name=platform_name)
```

Comment:
Could we please either use keyword-only args instead of multiple nameless `[]` parameters, or add mypy types (and add ) to disambiguate? See https://github.com/pantsbuild/pants/issues/7076. Adding mypy to this file requires adding the tag `partially-type-checked` to the owning target.

Reply:
I feel like adding partial type checking should be a separate change. I could make that and rebase this one on it, or do it after.
I'm happy to kwarg-ify the callsites I'm modifying as part of this change though.

Reply:
Sorry, could we also make the arguments themselves require keywords at every callsite? See #7076.

Reply:
Done.